pkg/k8shandler/logstore.go

```diff
@@ -97,51 +122,69 @@ func getElasticsearchCR(logging *logging.ClusterLogging, elasticsearchName strin
 	return cr
 }
 
-func createOrUpdateElasticsearchCR(logging *logging.ClusterLogging) error {
+func createOrUpdateElasticsearchCR(logging *logging.ClusterLogging) (err error) {
 
 	if utils.AllInOne(logging) {
 		esCR := getElasticsearchCR(logging, "elasticsearch")
 
-		err := sdk.Create(esCR)
+		err = sdk.Create(esCR)
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("Failure creating Elasticsearch CR: %v", err)
 		}
+
+		if err = updateElasticsearchCRIfRequired(esCR); err != nil {
+			return
+		}
 	} else {
 		esCR := getElasticsearchCR(logging, "elasticsearch-app")
 
-		err := sdk.Create(esCR)
+		err = sdk.Create(esCR)
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("Failure creating Elasticsearch CR: %v", err)
 		}
 
+		if err = updateElasticsearchCRIfRequired(esCR); err != nil {
+			return
+		}
+
 		esInfraCR := getElasticsearchCR(logging, "elasticsearch-infra")
 
 		err = sdk.Create(esInfraCR)
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("Failure creating Elasticsearch Infra CR: %v", err)
 		}
-	} /*else if errors.IsAlreadyExists(err) {
-	  // Get existing configMap to check if it is same as what we want
-	  existingCR := &v1alpha1.Elasticsearch{
-	    ObjectMeta: metav1.ObjectMeta{
-	      Name: "elasticsearch",
-	      Namespace: logging.Namespace,
-	    },
-	    TypeMeta: metav1.TypeMeta{
-	      Kind: "Elasticsearch",
-	      APIVersion: "logging.openshift.io/v1alpha1",
-	    },
-	  }
-
-	  err = sdk.Get(existingCR)
-	  if err != nil {
-	    logrus.Fatalf("Unable to get Elasticsearch CR: %v", err)
-	  }
-
-	  logrus.Infof("Found existing CR: %v", existingCR)
-
-	  // TODO: Compare existing CR labels, selectors and port
-	}*/
+
+		if err = updateElasticsearchCRIfRequired(esInfraCR); err != nil {
+			return
+		}
+	}
 
 	return nil
 }
+
+func updateElasticsearchCRIfRequired(desired *v1alpha1.Elasticsearch) (err error) {
+	current := desired.DeepCopy()
+
+	if err = sdk.Get(current); err != nil {
+		return fmt.Errorf("Failed to get Elasticsearch CR: %v", err)
+	}
+
+	current, different := isElasticsearchCRDifferent(current, desired)
+
+	if different {
+		if err = sdk.Update(current); err != nil {
+			return fmt.Errorf("Failed to update Elasticsearch CR: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func isElasticsearchCRDifferent(current *v1alpha1.Elasticsearch, desired *v1alpha1.Elasticsearch) (*v1alpha1.Elasticsearch, bool) {
+
+	different := false
+
+	//TODO: populate this
```

Comment:
Is this something that should be implemented?

Reply:
This will be implemented in 4.1
For now this is just a placeholder for elasticsearch